### DISCUSSION TITLE
Handling multiple requests

### DISCUSSION THREAD

**Discussion by dpolamarasetty** [mediagis/nominatim-docker#408] [General]
Hi, I have my own nominatim server running in AWS using the nominatim docker image. I have millions of addresses i need to verify, so i use the search method, from my understanding we cannot do batch processing using the search method. The only batch process allowed is if i already have the osm_ids. What is the best way to make sure my server can run parallel requests?and the best configurations for it? Should I be using nginix instead of apache ?Any suggestions?

**Comment by leonardehrenfried** [comment]
The docker image allows you to send parallel requests by default. Have you tried that?

how much paralellism depends on your hardware. once you've reached the limits of that you could  play around with the configuration. However, my gut feeling is that you will gain more performance fine tuning the Postgres config rather than switching to nginx.

This however very much a trial and error process.

I would be very interested in hearing about your findings so if you've gained any insights, please post them here.

**Reply by dpolamarasetty** [reply]
Right now, using the time diff before and after request load, it averages to 2 request per second. Server configuration is t2.2xlarge, vCPUs 8, volume 500GB,32GB in memory.

Can you tell me more about the Postgres fine tuning to query the data faster?

**Reply by leonardehrenfried** [reply]
Have you tried sending parallel requests? With 8 cores you should be able to achieve ~8 parallel requests.

I can't give you hints on the PG tuning. You're going to have to use a general tuning guide from elsewhere.
